common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -110,15 +123,16 @@ def test_page_view_event(self):
         And I visit my account settings page
         Then a page view analytics event should be recorded
         """
-        self.verify_events_of_type(
-            self.username,
-            u"edx.user.settings.viewed",
-            [{
-                u"user_id": long(self.user_id),
-                u"page": u"account",
-                u"visibility": None,
-            }]
-        )
+
+        event_filter = {
+            'event_type':'edx.user.settings.viewed',
+            'event': {
+                'user_id': self.user_id,
+                'page': 'account',
+                'visibility': None
+            }
+        }
+        self.wait_for_events(event_filter=event_filter, number_of_matches=1)
```

Comment:
Ah, excellent catch on the failure messages, I've detailed the revised failure messages below.

When the filter is a function:

```
    /edx/app/edxapp/venvs/edxapp/src/bok-choy/bok_choy/promise.py line 96 in fulfill
      raise BrokenPromise(self)
   BrokenPromise: Promise not satisfied: Waiting for 1 events to match the filter:
   /edx/app/edxapp/edx-platform/common/test/acceptance/tests/lms/test_account_settings.py:127
   --------------------
           def foo(e):
               return e['event_type'] == 'edx.user.settings.viewed2'
   --------------------
   
   Events Collected:
   
   {u'accept_language': u'en-US,en;q=0.5',
    u'agent': u'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:28.0) Gecko/20100101 Firefox/28.0',
    u'context': {u'course_id': u'',
                 u'org_id': u'',
                 u'path': u'/account/settings',
                 u'user_id': 22L},
    u'event': u'{"POST": {}, "GET": {}}',
    u'event_source': u'server',
    u'event_type': u'/account/settings',
    u'host': u'precise64',
    u'ip': u'127.0.0.1',
    u'page': None,
    u'referer': u'',
    u'time': datetime.datetime(2015, 4, 30, 21, 11, 55, 834000),
    u'username': u'test_173032'}
    ...
```

When it's a lambda:

```
    /edx/app/edxapp/venvs/edxapp/src/bok-choy/bok_choy/promise.py line 96 in fulfill
      raise BrokenPromise(self)
   BrokenPromise: Promise not satisfied: Waiting for 1 events to match the filter:
   /edx/app/edxapp/edx-platform/common/test/acceptance/tests/lms/test_account_settings.py:156
   --------------------
               event_filter=lambda e: e['event_type'] == 'edx.user.settings.viewed2', number_of_matches=1)
   --------------------
   
   Events Collected:
   
   {u'accept_language': u'en-US,en;q=0.5',
    u'agent': u'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:28.0) Gecko/20100101 Firefox/28.0',
    u'context': {u'course_id': u'',
                 u'org_id': u'',
                 u'path': u'/account/settings',
                 u'user_id': 23L},
    u'event': u'{"POST": {}, "GET": {}}',
    ...
```

And when it's just a dict:

```
    /edx/app/edxapp/venvs/edxapp/src/bok-choy/bok_choy/promise.py line 96 in fulfill
      raise BrokenPromise(self)
   BrokenPromise: Promise not satisfied: Waiting for 1 events to match the filter:
   --------------------
   {'event_type': 'edx.user.settings.viewed2'}
   --------------------
   
   Events Collected:
   
   {u'accept_language': u'en-US,en;q=0.5',
    u'agent': u'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:28.0) Gecko/20100101 Firefox/28.0',
    u'context': {u'course_id': u'',
                 u'org_id': u'',
                 u'path': u'/account/settings',
                 u'user_id': 24L},
    u'event': u'{"POST": {}, "GET": {}}',
    u'event_source': u'server'
    ...
```